File: src/runtime/dart/std/os_path_native.dart
// std/os_path_native.dart — stub native for pytra.std.os_path
// source: src/runtime/dart/std/os_path_native.dart

import 'dart:io';

String join(String a, String b) => '$a/$b';
String dirname(String p) {
  int idx = p.lastIndexOf('/');
  return idx < 0 ? '' : p.substring(0, idx);
}
String basename(String p) {
  int idx = p.lastIndexOf('/');
  return idx < 0 ? p : p.substring(idx + 1);
}
List<String> splitext(String p) {
  int idx = p.lastIndexOf('.');
  return idx < 0 ? [p, ''] : [p.substring(0, idx), p.substring(idx)];
}
String abspath(String p) => File(p).absolute.path;
bool exists(String p) => File(p).existsSync() || Directory(p).existsSync();

File: src/runtime/zig/std/os_path_native.zig
// Native implementation of pytra.std.os_path for Zig.
// source: src/pytra/std/os_path.py

const std = @import("std");

pub fn join(a: []const u8, b: []const u8) []const u8 {
    const alloc = std.heap.page_allocator;
    return std.fs.path.join(alloc, &[_][]const u8{ a, b }) catch "";
}

pub fn dirname(p: []const u8) []const u8 {
    return std.fs.path.dirname(p) orelse "";
}

pub fn basename(p: []const u8) []const u8 {
    return std.fs.path.basename(p);
}

pub fn splitext(p: []const u8) struct { _0: []const u8, _1: []const u8 } {
    const base = basename(p);
    const dot = std.mem.lastIndexOfScalar(u8, base, '.') orelse return .{ ._0 = p, ._1 = "" };
    const prefix_len = p.len - base.len + dot;
    return .{ ._0 = p[0..prefix_len], ._1 = base[dot..] };
}

pub fn abspath(p: []const u8) []const u8 {
    return p;
}

pub fn exists(p: []const u8) bool {
    _ = std.fs.cwd().statFile(p) catch return false;
    return true;
}
